src/main/java/mate/academy/springboot/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.springboot.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String field;
+    private String fieldMatch;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    @Override
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        Object fieldValue = new BeanWrapperImpl(value)
+                .getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(value)
+                .getPropertyValue(fieldMatch);
+
+        if (fieldValue != null) {
+            return fieldValue.equals(fieldMatchValue);
+        } else {
+            return fieldMatchValue == null;
+        }
```

Comment:
you can use the ternary operator here 
```suggestion
        return fieldValue != null && fieldValue.equals(fieldMatchValue);
```